test/primitives/kernels/test_new_vs_old_kernel.py

```diff
@@ -0,0 +1,90 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Test primitive-based quantum kernel versus the original implementation."""
+
+from __future__ import annotations
+
+import itertools
+
+from test import QiskitMachineLearningTestCase
+
+import numpy as np
+from ddt import ddt, idata, unpack
+from qiskit import BasicAer
+from qiskit.circuit.library import ZFeatureMap, ZZFeatureMap
+from qiskit.primitives import Sampler
+from qiskit.utils import algorithm_globals, QuantumInstance
+
+from qiskit_machine_learning.kernels import QuantumKernel as QuantumKernelOld
+from qiskit_machine_learning.primitives.kernels import QuantumKernel
+
+
+@ddt
+class TestNewVsOldQuantumKernel(QiskitMachineLearningTestCase):
```

Comment:
We already discussed this, but I really like this idea of unit test, and I plan to formally add it to the QNNs too :)